src/GraphQL.MicrosoftDI/GraphQLBuilder.cs

```diff
@@ -16,12 +16,15 @@ namespace GraphQL.MicrosoftDI
     /// An implementation of <see cref="IGraphQLBuilder"/> which uses the Microsoft dependency injection framework
     /// to register services and configure options.
     /// </summary>
-    public class GraphQLBuilder : GraphQLBuilderBase, IServiceCollection
+    public class GraphQLBuilder : GraphQLBuilderBase, IServiceCollection, IServiceRegister
     {
+        /// <inheritdoc />
+        public override IServiceRegister Services => this;
+
         /// <summary>
         /// Returns the underlying <see cref="IServiceCollection"/> of this builder.
         /// </summary>
-        public IServiceCollection Services { get; }
+        public IServiceCollection ServiceCollection { get; }
```

Comment:
`GraphQLBuilder` implements `IServiceCollection` and at the same time has ` public IServiceCollection ServiceCollection { get; }` . Are they both required?

Reply:
It implements `IServiceCollection` so that the server project can cast it to `IServiceCollection` without needing a reference to `GraphQL.MicrosoftDI`.  (Since to use the ASP.Net authorize extension, it needs access to the original service collection.)

The other design pattern would be for the server project to cast to `GraphQL.MicrosoftDI.GraphQLBuilder` and then access the `Services` property.  However, this means that it would not support any other Microsoft DI implementation unless it derives from `GraphQL.MicrosoftDI.GraphQLBuilder`.

I opted for the former choice.  The `Services` property exists as a convenience for anyone using the class directly.

Reply:
The third choice would be to add an interface within the `GraphQL.MicrosoftDI` package that encapsulates the `Services` property, for example `IMicrosoftServices` with one property `ServiceCollection` which returns the `IServiceCollection`.  But this seems a bit pointless.  Might as well just cast to `IServiceCollection`.

Reply:
Are you OK with name `ServiceCollection`? You wanted `Services` as the name for `IServiceRegister` property so I had to choose another name for this property.

Reply:
Yes, perfect.